lib/error.g

```diff
@@ -59,16 +59,58 @@ end);
 
 ErrorLVars := fail;
 
-BIND_GLOBAL("WHERE", function( context, depth, outercontext)
-    local   bottom,  lastcontext,  f;
+BIND_GLOBAL("PRETTY_PRINT_VARS", function(context)
+    local vars, i, argcount, val;
+    vars := ContentsLVars(context);
+    if IsRecord(vars) then
+        argcount := ABS_RAT(NumberArgumentsFunction(vars.func));
+        PrintTo(ERROR_OUTPUT, "\>\>\n\<\<arguments:");
+        if argcount = 0 then
+            PrintTo(ERROR_OUTPUT, " <none>");
+        else
+            for i in [1..argcount] do
+                if IsBound(vars.values[i]) then
+                    val := vars.values[i];
+                else
+                    val := "<unassigned>";
+                fi;
+                # This must all be on one line as PrintTo(ERROR_OUTPUT,...) resets indentation on every call
+                PrintTo(ERROR_OUTPUT, "\>\>\>\>\n", vars.names[i], " :=\>\> ", val, "\<\<\<\<\<\<");
+            od;
+        fi;
+
+        PrintTo(ERROR_OUTPUT, "\>\>\n\<\<local variables:");
+        if argcount = Length(vars.names) then
+            PrintTo(ERROR_OUTPUT, " <none>");
+        else
+            for i in [argcount+1..Length(vars.names)] do
+                if IsBound(vars.values[i]) then
+                    val := vars.values[i];
+                else
+                    val := "<unassigned>";
+                fi;
+                # This must all be on one line as PrintTo(ERROR_OUTPUT,...) resets indentation on every call
+                PrintTo(ERROR_OUTPUT, "\>\>\>\>\n", vars.names[i], " :=\>\> ", val, "\<\<\<\<\<\<");
+            od;
+        fi;
+        PrintTo(ERROR_OUTPUT,"\n");
```

Comment:
This leaves the line break hints unbalanced, which we shouldn't do (see also issue #2677)

Reply:
I think they are balanced? (in fact, every PrintTo is balanced)

Reply:
My apologies, Mi miscounted. This is all fine!